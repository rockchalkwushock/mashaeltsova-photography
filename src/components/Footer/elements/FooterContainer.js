import styled from 'styled-components'

const FooterContainer = styled.div`
  text-align: center;
  @media (min-width: ${({ theme }) => theme.sizes.md}) {
    text-align: ${({ left }) => (left ? 'left' : 'right')};
  }
`

FooterContainer.displayName = 'FooterContainer'

export default FooterContainer
