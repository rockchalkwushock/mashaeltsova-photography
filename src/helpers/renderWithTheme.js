import React from 'react'
import { mount } from 'enzyme'
import { ThemeProvider } from 'styled-components'
import theme from '../utils/theme'

/**
 * @fileOverview
 *
 * HOC for testing components with `theme` prop being used.
 */

const renderWithTheme = Component =>
  mount(<ThemeProvider theme={theme}>{Component}</ThemeProvider>)

export default renderWithTheme
