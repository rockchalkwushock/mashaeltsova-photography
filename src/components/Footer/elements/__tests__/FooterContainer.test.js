import React from 'react'

import { renderWithTheme } from '../../../../helpers'
import FooterContainer from '../FooterContainer'

describe('Styled Component: <FooterContainer />', () => {
  let wrapper
  let wrapper2
  beforeAll(() => {
    wrapper = renderWithTheme(<FooterContainer />)
    wrapper2 = renderWithTheme(<FooterContainer left />)
  })
  test('should render markup', () => {
    expect(wrapper).toMatchSnapshot()
    expect(wrapper2).toMatchSnapshot()
  })
  test('should have styles', () => {
    expect(wrapper).toHaveStyleRule('text-align', 'center')
    expect(wrapper2).toHaveStyleRule('text-align', 'center')
  })
  test('should have styles as right when bigger viewport', () => {
    expect(wrapper).toHaveStyleRule('text-align', 'right', {
      media: '(min-width:736px)'
    })
  })
  test('should have styles as left when bigger viewport', () => {
    expect(wrapper2).toHaveStyleRule('text-align', 'left', {
      media: '(min-width:736px)'
    })
  })
})
