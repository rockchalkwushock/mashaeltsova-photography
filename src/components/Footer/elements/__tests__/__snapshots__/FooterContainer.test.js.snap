// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Styled Component: <FooterContainer /> should render markup 1`] = `
.c0 {
  text-align: center;
}

@media (min-width:736px) {
  .c0 {
    text-align: right;
  }
}

<ThemeProvider
  theme={
    Object {
      "sizes": Object {
        "lg": "1000px",
        "md": "736px",
        "sm": "540px",
      },
    }
  }
>
  <FooterContainer>
    <div
      className="c0"
    />
  </FooterContainer>
</ThemeProvider>
`;

exports[`Styled Component: <FooterContainer /> should render markup 2`] = `
.c0 {
  text-align: center;
}

@media (min-width:736px) {
  .c0 {
    text-align: left;
  }
}

<ThemeProvider
  theme={
    Object {
      "sizes": Object {
        "lg": "1000px",
        "md": "736px",
        "sm": "540px",
      },
    }
  }
>
  <FooterContainer
    left={true}
  >
    <div
      className="c0"
    />
  </FooterContainer>
</ThemeProvider>
`;
